### PR TITLE
feat: Dynamically generate manifest JSON

### PR DESCRIPTION
**Pull request by munen**
Whenever the user bookmarks to the homescreen, the manifest.json will be read and the `start_url` property will be used for the actual bookmark URL. 

Before, we had a hard coded "/index.html". iOS ignored the `start_url` attribute, so bookmarking specific files still worked. Android already read this attribute, so there was extra work involved to bookmark a specific file as a PWA.

With this change, using any file as an entry in the PWA mode should work.

Tested on an iPhone Xs with the latest stable iOS.

Thank you @aspiers for introducing me to this option in https://github.com/200ok-ch/organice/pull/463 :pray:

### CODE REVIEW

File: public/index.html [hunk 1]
```diff
@@ -45,5 +45,38 @@
...
+                 "sizes": "192x192",
+                 "type": "image/png"
+             },
+             {
+                 "src": "favicon.ico",
+                 "sizes": "64x64 32x32 24x24 16x16",
+                 "type": "image/x-icon"
+             }
+         ],
+         "start_url": `${pathname || "/index.html"}`,
```
Comment (aspiers) [inline public/index.html:71]: No need for the template literal here?

```suggestion
         "start_url": pathname || "/index.html",
```

**Review by aspiers — APPROVED** [review]
Looks great, modulo the nitpick about interpolation. Thanks!

File: public/index.html (hunk 1 above)
Comment (munen) [reply public/index.html]: :+1:

**Comment by aspiers** [comment]
This is a subtle but really awesome new feature.  So I think it's definitely worth documenting too :-)

**Comment by munen** [comment]
> This is a subtle but really awesome new feature.

Yes, I think so, too. Thank you for pointing me to the right JavaScript API! I think I've never heard about it. On the other hand, most applications I worked on have a back-end, so they wouldn't necessarily have the need for this kind of API. But still, to have never heard of it feels a little off. I'll certainly tell other people about it, because it's a nice API^^

>  So I think it's definitely worth documenting too :-)

Already on it^^

**Comment by munen** [comment]
@aspiers What do you think about this doc change: 727f53f

**Comment by aspiers** [comment]
[**@munen**](https://github.com/munen) commented on [September 7, 2020 3:10 PM](https://github.com/200ok-ch/organice/pull/465#issuecomment-688351152 "2020-09-07T14:10:44Z - Replied by Github Reply Comments"):
> > This is a subtle but really awesome new feature.
> 
> Yes, I think so, too. Thank you for pointing me to the right JavaScript API! I think I've never heard about it. On the other hand, most applications I worked on have a back-end, so they wouldn't necessarily have the need for this kind of API. But still, to have never heard of it feels a little off. I'll certainly tell other people about it, because it's a nice API^^

Yes it's pretty obscure I guess - at least I'd never heard about it either, until [I started researching dynamic manifests for adding share target support](https://github.com/200ok-ch/organice/issues/52#issuecomment-657092025) :-)

**Comment by aspiers** [comment]
[**@munen**](https://github.com/munen) commented on [September 7, 2020 3:14 PM](https://github.com/200ok-ch/organice/pull/465#issuecomment-688353154 "2020-09-07T14:14:29Z - Replied by Github Reply Comments"):
> [@aspiers](https://github.com/aspiers) What do you think about this doc change: [727f53f](https://github.com/200ok-ch/organice/commit/727f53fb80b5e58e00c3de432a9f8baf0a000b5d) ?

Looks perfect! :+1:

**Comment by munen** [comment]
> Looks perfect! +1

Sweet!

I've updated the changelog and will merge.

**Comment by aspiers** [comment]
I already posted this in #52, but leaving here too for posterity:

In case it's relevant, it looks like [How Chrome handles updates to the web app manifest](https://web.dev/manifest-updates/) has some useful info, and maybe [Document how other browsers handle web app manifest updates · Issue #4038 · GoogleChrome/web.dev](https://github.com/GoogleChrome/web.dev/issues/4038) will accumulate some too.

**Pull request merged**